### DISCUSSION TITLE
curl change to use SecItemCopyMatching with kSecAttrApplicationLabel instead of kSecAttrLabel.

### DISCUSSION THREAD

**Discussion by securitywizard** [curl/curl#6343] [General]
Hi 

I have a Yubikey PIV card that I use with curl to authenticate to my services from macOS, as macOS changed to its own cryptotokenkit I can't use my Yubikey cert as curl is looking for "kSecAttrLabel" but which is not supported for smart cards on Mac anymore, so I want it to find public key hash of the key that I want it to use which can be possible if  "kSecAttrLabel" is replaced by "kSecAttrApplicationLabel".

I want to explore by options here instead of treating this as bug as the traditional certificate on the keychain can still be queried using the CN/Label (kSecAttrLabel). I want to see if I can compile a version of curl with this particular change to "sectransp.c".

I'm sure that there might be many options that I'm not exploring here, please give your suggestions on how to resolve the issue.

Thanks,
Manoj.

**Comment by bagder** [comment]
I don't think asking this very narrow and niche question here will reach anyone who knows the answer. Secure Transport is an esoteric backend, abandoned by Apple and known by very few curl developers.
